app/policies/adopter_application_policy.rb

```diff
@@ -0,0 +1,33 @@
+class AdopterApplicationPolicy < ApplicationPolicy
+  authorize :pet, optional: true
+
+  pre_check :verify_adopter_profile!, only: %i[index? create?]
+  pre_check :verify_pet_adoptable!, only: %i[create?]
+
+  def update?
+    applicant? && permission?(:withdraw_adopter_applications)
+  end
+
+  def index?
```

Comment:
should this also check for applicant? to make sure a user only sees their applications.

Reply:
The checking for the applicant ownership for `index?` is actually done in the `relation_scope` instead of in `index?`. So far, I have used scoping in policies only for index actions for things like this. The scope is missing in this commit though. I added the scope in the next commit once I realized it was missing in this first iteration. c0e41c5

Reply:
Ahh I see ok thank you for clarifying that.